### PR TITLE
Add interactive command to get a problem from a category

### LINKED ISSUE #80: Add interactive command to get a random problem from a category
Categories include: Array, Hashmap, Dynamic Programming, etc.

### PR DESCRIPTION
**Pull request by kajahno**
Closes #80

### CODE REVIEW

File: api/leetcode-data.js [hunk 1]
```diff
@@ -76,29 +94,30 @@ export async function getWeeklyProblem() {
...
-        "Simulation",
-        "Sliding Window",
-        "Sorting",
-        "Stack",
-        "String Matching",
-        "String",
-        "Trie",
-        "Two Pointers",
-    ];
+    // { "category": "slug or ID" }
```
Comment (nramirez) [inline api/leetcode-data.js:97]: do you need this comment?

File: api/leetcode.js [hunk 2]
```diff
@@ -94,6 +97,30 @@ export class Leetcode {
...
+    getRandomProblemFromCategoryMessage = async (category) => {
+        const randomProblemObj = await getRandomProblemFromCategory(category);
+
+        if (!randomProblemObj) {
+            logger.error("Unable to fetch problemSet");
+            return;
+        }
+
+        const categories = await getProblemCategories();
+        const categoryName = Object.keys(categories).find((key) => categories[key] === category);
```
Comment (nramirez) [inline api/leetcode.js:112]: interesting, maybe I'm missing something here, why is it not possible to do `categories[key]` ?

File: commands.js [hunk 3]
```diff
@@ -59,20 +53,46 @@ function createCommandChoices() {
...
 
+// Get the challenge categories from api/leetcode-data.js
+export async function challengeCategoriesCommandChoices() {
+    const commandChoices = [];
+    for (const [key, value] of Object.entries(await getProblemCategories())) {
+        commandChoices.push({
+            name: key,
+            value,
+        });
+    }
```
Comment (nramirez) [inline commands.js:64]: you could do this mapping in one line 
```js
const commandChoices = Object.entries(await getProblemCategories()).map(([key, value]) => ({
    name: key,
    value,
}));

```
or better yet, you could probably do `new Map(categories)`

**Review by nramirez — APPROVED** [review]
Looks good! 🙌

File: api/leetcode.js (hunk 2 above)
Comment (kajahno) [reply api/leetcode.js:112]: It's because the key is actually the value..I'll swap in the data structure and do the lookup as you mention.

**Pull request merged**